chore(ci): define Github labels as code with a workflow

qdm12 opened the pull request:
## Why this should be merged

Github issues are currently using the default github labels, and it would be good to have custom and easily customizable-as-code Github labels to better categorize Github issues.

## How this works

- New workflow `labels.yml` using [crazy-max/ghaction-github-labeler](https://github.com/crazy-max/ghaction-github-labeler) to have Github labels as code in `.github/labels.yml`
- Custom labels defined in `.github/labels.yml`. Note:
  - default old ones will all be removed, except migrated ones below:
  - `bug` label will be migrated to `Category: Bug 🐛` label
  - `DO NOT MERGE` label will be migrated to `Status: 🔴 DO NOT MERGE`
- Dry run the labels job on pull requests affecting labels configuration files
- New workflow `labels-skip.yml` to satisfy required workflow passing for pull requests (not set yet but should be set soon), triggering only on pull requests and when none of the labels related files are modified. This can be removed eventually as well, since we don't really need to make the labels workflow a required workflow, but I think it's a good-to-know at least during the PR review 😉  Also note it runs on every commit here since `.github/workflows/labels-skip.yml` is not part of the ignored paths.

## How this was tested

Dry run on this pull request

qdm12 committed:
chore(ci): define github labels in code

qdm12 committed:
Add DO NOT MERGE label migration

ARR4N commented on .github/workflows/label-skip.yml:
The things we have to do to work around GH shortcomings 🤦 🤣 

I think I'd prefer to have the `labels` job just not be required than have a workaround like this. The only time we care about it running is when we want labels updated (rare to begin with), and then it's front of mind to check that it was triggered. While it's nice to have full automation, I don't think that it's warranted on this occasion (although very handy trick to know about for more critical jobs, TY!).

qdm12 replied:
Yeah I thought about adding it for knowledge sharing purposes really 😉  Removed!

ARR4N commented on .github/labels.yml:
```diff
@@ -0,0 +1,71 @@
+  description: "Unit tests, integration tests, etc."
+  color: "ffc7ea"
+- name: "Category: Code quality ✏️"
+  description: "Unit tests, integration tests, etc."
```
This looks like an accidental copy-paste.

ARR4N commented on .github/labels.yml:
```diff
@@ -0,0 +1,71 @@
+# Complexity
+- name: "Complexity: 🟥 Hard to do"
+  color: "ff9efc"
+- name: "Complexity: 🟩 Easy to do"
```
Is there a blue square for easy (think hot vs cold)? Red-green isn't great for colourblindness and IIRC red-blue is safe for all types.

qdm12 replied:
> colourblindness

Next level 🫡 ! Blue it is!

ARR4N commented on .github/labels.yml:
```diff
@@ -0,0 +1,71 @@
+- name: "Closed: ⚰️ Inactive"
+  color: "959a9c"
+  description: "No answer was received for weeks"
+- name: "Closed: 🗑️ Bad issue"
```
When would this be used? Feels a bit passive aggressive 😬

qdm12 replied:
😄 It does, it would be for example if the issue is reported but it's irrelevant to the project (i.e. internet does not work for the user). Now as a library, we could probably remove that one since we won't have much of the public creating "bad issues", it'll mostly be internal first. REMOVED!

qdm12 committed:
Remove label-skip workflow

qdm12 committed:
Remove invalid code quality description line

qdm12 committed:
Change complexity green square to blue square

qdm12 committed:
Remove Closed: bad issue label

ARR4N reviewed (approved):
Approved pending https://github.com/ava-labs/libevm/pull/95 into this branch as it's easier than a review.

ARR4N committed:
chore: migrate instead of delete old labels (#95)

Easier than PR suggestions. I noticed that some useful labels were being
deleted either entirely or to be replaced with effectively the same
thing.

---------

Signed-off-by: Arran Schlosberg <519948+ARR4N@users.noreply.github.com>
Co-authored-by: Quentin McGaw <quentin.mcgaw@gmail.com>
